deluge/tests/test_httpdownloader.py

```diff
@@ -154,140 +156,136 @@ def setUp(self):  # NOQA
-        d = download_file(url, fname('forced'), force_filename=True)
-        d.addCallback(self.assertEqual, fname('forced'))
-        d.addCallback(self.assertContains, 'This file should be called spam')
-        return d
+        filename = await download_file(url, fname('forced'), force_filename=True)
+        assert filename == fname('forced')
+        self.assert_contains(filename, 'This file should be called spam')
 
-    def test_download_with_gzip_encoding(self):
+    @pytest_twisted.ensureDeferred
+    async def test_download_with_gzip_encoding(self):
         url = self.get_url('gzip?msg=success')
-        d = download_file(url, fname('gzip_encoded'))
-        d.addCallback(self.assertContains, 'success')
-        return d
+        filename = await download_file(url, fname('gzip_encoded'))
+        self.assert_contains(filename, 'success')
 
-    def test_download_with_gzip_encoding_disabled(self):
+    @pytest_twisted.ensureDeferred
+    async def test_download_with_gzip_encoding_disabled(self):
         url = self.get_url('gzip?msg=unzip')
-        d = download_file(url, fname('gzip_encoded'), allow_compression=False)
-        d.addCallback(self.assertContains, 'unzip')
-        return d
+        filename = await download_file(
+            url, fname('gzip_encoded'), allow_compression=False
+        )
+        self.assert_contains(filename, 'unzip')
 
-    def test_page_redirect_unhandled(self):
+    @pytest_twisted.ensureDeferred
+    async def test_page_redirect_unhandled(self):
         url = self.get_url('redirect')
-        d = download_file(url, fname('none'))
-        d.addCallback(self.fail)
+        with pytest.raises(PageRedirect):
+            await download_file(url, fname('none'), handle_redirects=False)
 
-        def on_redirect(failure):
-            self.assertTrue(type(failure), PageRedirect)
+    @pytest_twisted.ensureDeferred
+    async def test_page_redirect(self):
+        url = self.get_url('redirect')
+        filename = await download_file(url, fname('none'), handle_redirects=True)
+        assert filename == fname('none')
 
-        d.addErrback(on_redirect)
-        return d
+    @pytest_twisted.ensureDeferred
+    async def test_page_not_found(self):
+        with pytest.raises(Error):
+            await download_file(self.get_url('page/not/found'), fname('none'))
 
-    def test_page_redirect(self):
-        url = self.get_url('redirect')
-        d = download_file(url, fname('none'), handle_redirects=True)
-        d.addCallback(self.assertEqual, fname('none'))
-        d.addErrback(self.fail)
-        return d
-
-    def test_page_not_found(self):
-        d = download_file(self.get_url('page/not/found'), fname('none'))
-        d.addCallback(self.fail)
-        d.addErrback(self.assertIsInstance, Failure)
-        return d
-
-    def test_page_not_modified(self):
+    @pytest.mark.xfail(reason="Doesn't seem like httpdownloader ever implemented this.")
```

Comment:
Not sure if we care about this or not, I didn't see any code that would actually handle this case.